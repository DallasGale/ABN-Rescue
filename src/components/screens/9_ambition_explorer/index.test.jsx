/* eslint-env jest */

import React from 'react'
import { render } from '@testing-library/react'

import AmbitionExplorer from '.'

const mockApi = {
  suburbs: [
    {
      postcode: '3777',
      name: 'Sydney City',
      cancelledAbns: '20784',
    },
    {
      postcode: '3778',
      name: 'Surry Hills',
      cancelledAbns: '3567',
    },
    {
      postcode: '3779',
      name: 'Paramatta',
      cancelledAbns: '1835',
    },
    {
      postcode: '3780',
      name: 'Vaucluse',
      cancelledAbns: '1020',
    },
    {
      postcode: '3781',
      name: 'Darling Point',
      cancelledAbns: '924',
    },
    {
      postcode: '3782',
      name: 'Wollahara',
      cancelledAbns: '584',
    },
  ],
}

const props = {
  apiData: mockApi,
  postcodeOnChange: () => { },
  ctaOnClick: () => { },
  ctaLabel: 'button label',
  ctaLink: '/link',
  value: 'value',
  postcodeFieldRef: React.createRef(),
  errorMsg: [],
}

describe('<AmbitionExplorer />', () => {
  it('renders without crashing', () => {
    const component = render(<AmbitionExplorer {...props} />)
    expect(component).toMatchSnapshot()
  })
})
