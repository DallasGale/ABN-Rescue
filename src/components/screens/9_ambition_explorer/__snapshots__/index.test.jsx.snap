// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AmbitionExplorer /> renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="styled__StyledWrapper-ey9dti-0 kpuNbG"
        style=""
      >
        <div
          class="scrollmagic-pin-spacer"
          data-scrollmagic-pin-spacer=""
          style="top: 0px; display: block; position: relative; box-sizing: content-box; width: 100%; min-height: 0px; height: auto; padding-top: 0px; padding-bottom: 7000px;"
        >
          <div
            class="sticky"
            style="position: relative; margin: auto; box-sizing: border-box; width: 100%;"
          >
            <div />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="styled__StyledWrapper-ey9dti-0 kpuNbG"
      style=""
    >
      <div
        class="scrollmagic-pin-spacer"
        data-scrollmagic-pin-spacer=""
        style="top: 0px; display: block; position: relative; box-sizing: content-box; width: 100%; min-height: 0px; height: auto; padding-top: 0px; padding-bottom: 7000px;"
      >
        <div
          class="sticky"
          style="position: relative; margin: auto; box-sizing: border-box; width: 100%;"
        >
          <div />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
